connection.go

```diff
@@ -362,8 +368,17 @@ func Connect(ctx context.Context, dialer Dialer, opts Opts) (conn *Connection, e
 
 	conn.cond = sync.NewCond(&conn.mutex)
 
-	if err = conn.createConnection(ctx); err != nil {
-		return nil, err
+	if conn.opts.Reconnect > 0 {
+		conn.mutex.Lock()
```

Comment:
Please write a comment here about the fact that a mutex is needed here for the simplicity of the `runReconnects` method implementation.

Reply:
Fixed.